src/components/Tooltip/Tooltip.js

```diff
@@ -0,0 +1,134 @@
+// @flow
+import React from 'react';
+import { Manager, Popper, Reference } from 'react-popper';
+import onClickOutside from 'react-onclickoutside';
+
+import { TooltipTargetTag, TooltipMessageTag } from './Tooltip.theme';
+
+
+type RenderProps = {
+  isOpen: boolean,
+  openTooltip: () => void,
+  closeTooltip: () => void,
+  toggleTooltip: () => void,
+}
+
+type TootipProps = {
+  children: React$Node | (RenderProps) => React$Node,
+  /** Default content position relative target */
+  placement?: 'top' | 'left' | 'bottom' | 'right',
+  /** Custom tag of the tooltip target */
+  tagName?: string,
+  /** Custom style of the cursor */
+  cursor?: string,
+  /** Tooltip message */
+  message: React$Node,
+  /** Default state of the opening */
+  defaultOpen?: boolean,
+  /** Possible tooltip trigger */
+  trigger?: 'hover' | 'click',
+}
+
+type TootipState = {
+  isOpen: boolean,
+}
+
+const Tooltip: React$ComponentType<TootipProps> = onClickOutside(
+  class Tooltip extends React.PureComponent<TootipProps, TootipState> {
+
+    static defaultProps = {
+      defaultOpen: false,
+      trigger: 'hover',
+      tagName: 'span',
+      cursor: 'default',
+    };
+
+    constructor(props: TootipProps) {
+      super(props);
+
+      this.state = {
+        isOpen: props.defaultOpen || false,
+      };
+    }
+
+    openTooltip =() => this.setState({ isOpen: true });
+    closeTooltip = () => this.setState({ isOpen: false });
+    toggleTooltip = () => this.setState({ isOpen: !this.state.isOpen });
+
+    handleClickOutside = (event: *) => {
```

Comment:
@ej9x Is this function used in the code?

Reply:
yes. part of usage react-onclickoutside